src/Chapter1.hs

```diff
@@ -542,7 +545,28 @@ value after "=" where the condition is true.
 Casual reminder about adding top-level type signatures for all functions :)
 -}
 
-mid x y z = error "mid: not implemented!"
+sortSortedLists :: [Int] -> [Int] -> [Int]
+sortSortedLists xs [] = xs
+sortSortedLists [] ys = ys
+sortSortedLists (x:xs) (y:ys)
+  | x <= y    = x : sortSortedLists xs (y:ys)
+  | otherwise = y : sortSortedLists (x:xs) ys
+
+
+mergeSort :: [Int] -> [Int]
+mergeSort [] = []
+mergeSort (x:[]) = [x]
+mergeSort (x:y:[]) = min x y : max x y : []
+mergeSort xs = do
+  let sortedFirstHalf = mergeSort (fst halves)
+  let sortedSecondHalf = mergeSort (snd halves)
+  sortSortedLists sortedFirstHalf sortedSecondHalf
+    where halves = splitAt (length xs `div` 2) xs
```

Comment:
That's a valid implementing of splitting the list into two halves. However, calculating the length of a list is a slow operation, as it traverses the whole list. In this particular case it is possible to split the list into two equal size lists in a single pass — elements on odd positions and elements on even positions.